### PR TITLE
Some bug fixes from my fork

### PR DESCRIPTION
**Pull request by ful1e5**
#### Fixes
Nested colors overide ful1e5/onedark.nvim#6
Minimal Status Line Bug ful1e5/onedark.nvim#2

### CODE REVIEW

File: lua/onedark/theme.lua [hunk 1]
```diff
@@ -354,14 +353,21 @@ function M.setup(config)
...
+    if vim.o.statusline ~= nil and string.find(vim.o.statusline, "lualine") then
+      -- Fix VertSplit & StatusLine crossover when lualine is active
+      -- https://github.com/ful1e5/onedark.nvim/issues/2
+      -- https://github.com/hoob3rt/lualine.nvim/issues/274
+      theme.base.StatusLine = { bg = c.bg }
+
+      -- LuaLine
+      for _, section in pairs({ "a", "b", "c" }) do
+        theme.plugins["lualine_" .. section .. "_inactive"] = inactive
+      end
```
Comment (monsonjeremy) [inline lua/onedark/theme.lua:370]: Was this the code that you changed in order to fix the `inactiveStatusLine` still showing a different background? 

I just tried this one my local and I'm still seeing the issue. I feel like it's somewhere else in my package. If I use `master` of your repository it works fine for me but as soon as I switch to mine the status line doesn't hide properly 😭

**Comment by ful1e5** [comment]
Now fixed

File: lua/onedark/util.lua [hunk 2]
```diff
@@ -209,53 +184,53 @@ function util.terminal(colors)
...
-  util.syntax(theme.plugins)
-  util.terminal(theme.colors)
-  util.autocmds(theme.config)
+
+  -- load syntax for plugins and terminal async
+  vim.defer_fn(function()
+    util.terminal(theme.colors)
+    util.syntax(theme.plugins)
+    util.autocmds(theme.config)
+  end, 0)
```
Comment (monsonjeremy) [inline lua/onedark/util.lua:209]: This is awesome, I believe this is the fix for the bug! Do you know why this works? I assume it prevents a race condition with overriding the `lualine` inactive config?

**Review by monsonjeremy — APPROVED** [review]
Thanks a ton for doing this. Looks great!

**Pull request merged**